### PR TITLE
autoconfigure properties for a CassandraVectorStore cql connection to AstraDB

### PR DESCRIPTION
**Pull request by michaelsembwever**
…

### CODE REVIEW

File: spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vectorstore/astra/AstraExtraSettings.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+spring.cassandra.keyspace-name=default_keyspace
+spring.cassandra.username=token
+spring.cassandra.password=${ASTRA_DB_APPLICATION_TOKEN}
+spring.ai.vectorstore.astra.secure-connect-bundle=${ASTRA_SCB_PATH} </code>
+ *
+ * @author Cédrick Lunven
+ * @since 1.0.0
+ */
+@ConfigurationProperties(prefix = "spring.ai.vectorstore.astra")
+public class AstraExtraSettings {
```
Comment (tzolov) [inline spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vectorstore/astra/AstraExtraSettings.java:36]: The convention for property class is to use the `Properties` suffix. Check the other auto-config properties for example.

File: spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vectorstore/astra/AstraExtraSettings.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+# Cassandra Configuration
+spring.cassandra.keyspace-name=default_keyspace
+spring.cassandra.username=token
+spring.cassandra.password=${ASTRA_DB_APPLICATION_TOKEN}
+spring.ai.vectorstore.astra.secure-connect-bundle=${ASTRA_SCB_PATH} </code>
+ *
+ * @author Cédrick Lunven
+ * @since 1.0.0
+ */
+@ConfigurationProperties(prefix = "spring.ai.vectorstore.astra")
```
Comment (tzolov) [inline spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vectorstore/astra/AstraExtraSettings.java:35]: for consistency  define a CONFIG_PREFIX constant.  Check the other property implementations for example.

File: spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vectorstore/astra/AstraExtraSettings.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+public class AstraExtraSettings {
+
+	@Nullable // null means vanilla CQL (i.e. AstraDB not enabled)
+	private File secureConnectBundle;
+
+	public File getSecureConnectBundle() {
+		return secureConnectBundle;
+	}
+
+	public AstraExtraSettings setSecureConnectBundle(File secureConnectBundle) {
```
Comment (tzolov) [inline spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vectorstore/astra/AstraExtraSettings.java:45]: Use a simple setter, e.g. void response.

File: spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vectorstore/astra/AstraConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import java.nio.file.Path;
+
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * @author Cédrick Lunven
+ * @since 1.0.0
+ */
+@Configuration
+public class AstraConfiguration {
```
Comment (tzolov) [inline spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vectorstore/astra/AstraConfiguration.java:31]: How this configuration is used with rest of the cassandra vector DB support?

**Review by tzolov — CHANGES_REQUESTED** [review]
Apart from the code change request I'm not sure how the AstraConfiguration is related to the CassandraVectorStore?  This is neither auto-configuraiton (don't see it in the AutoConfiguration.imports) nor is used by the  other autoconfiguraitons? 
Also I'm missing documentation, and  ITs.

File: spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vectorstore/astra/AstraConfiguration.java (hunk 4 above)
Comment (michaelsembwever) [reply spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vectorstore/astra/AstraConfiguration.java:31]: AstraDB requires the secureConnectBundle for its CQL connection using a cassandra driver to work. this layers on top of all other Cassandra options.

I will write up more docs, and address the other comments…

**Pull request closed without merging**